### ISSUE TITLE
gradle-experimental:0.6.0-alpha7 with minifyEnabled = true does not work

### ISSUE THREAD

**Issue by zlwu** [googlesamples/android-ndk#146]
Andy ndk project using gradle-experimental:0.6.0-alpha7 with following gradle script will sync fail with message "Error:Exception thrown while executing model rule: BaseComponentModelPlugin.Rules#createBinaryTasks".

```gradle
apply plugin: 'com.android.model.application'

model {
  // ..
  android.buildTypes {
    release {
      minifyEnabled true
      proguardFiles.add(file('proguard-rules.txt'))
    }
  }
}
```

**Comment by ph0b** [comment]
The error returned is quite cryptic, but I think you're just missing the = to do the assignment on minifyEnabled:

```
apply plugin: 'com.android.model.application'

model {
  // ..
  android.buildTypes {
    release {
      minifyEnabled = true
      proguardFiles.add(file('proguard-rules.txt'))
    }
  }
}
```

**Comment by zlwu** [comment]
Actually the '=' is not necessary here. The issue still exist when the '=' is added.

**Comment by ph0b** [comment]
ok. Indeed the syntax without the = seems to work again here, sorry for the false lead.

I did reproduce the issue. When setting it to false, the build is fine btw.

I ran gradle with --stracktrace to get slightly more information:
Caused by: java.lang.NullPointerException
        at com.android.build.gradle.managed.adaptor.BuildTypeAdaptor.isUseProguard(BuildTypeAdaptor.java:113)
        at com.android.build.gradle.internal.TaskManager.doCreateMinifyTransform(TaskManager.java:2682)
        at com.android.build.gradle.internal.TaskManager.createMinifyTransform(TaskManager.java:2667)
        at com.android.build.gradle.internal.TaskManager.createPostCompilationTasks(TaskManager.java:1933)
        at com.android.build.gradle.internal.ApplicationTaskManager$9.call(ApplicationTaskManager.java:178)
        at com.android.build.gradle.internal.ApplicationTaskManager$9.call(ApplicationTaskManager.java:167)
        at com.android.builder.profile.ThreadRecorder$1.record(ThreadRecorder.java:55)
        at com.android.builder.profile.ThreadRecorder$1.record(ThreadRecorder.java:47)
        at com.android.build.gradle.internal.ApplicationTaskManager.createTasksForVariantData(ApplicationTaskManager.java:166)
        at com.android.build.gradle.internal.VariantManager.createTasksForVariantData(VariantManager.java:450)
        at com.android.build.gradle.model.BaseComponentModelPlugin$Rules.createBinaryTasks(BaseComponentModelPlugin.java:597)
        at org.gradle.model.internal.method.WeaklyTypeReferencingMethod.invoke(WeaklyTypeReferencingMethod.java:96)
        at org.gradle.model.internal.inspect.DefaultModelRuleInvoker.invoke(DefaultModelRuleInvoker.java:37)
        at org.gradle.model.internal.inspect.MethodBackedModelAction.execute(MethodBackedModelAction.java:43)
        at org.gradle.model.internal.core.AbstractModelActionWithView.execute(AbstractModelActionWithView.java:46)
        at org.gradle.model.internal.registry.DefaultModelRegistry$2.run(DefaultModelRegistry.java:483)
        at org.gradle.model.internal.registry.RuleContext.run(RuleContext.java:59)
        at org.gradle.model.internal.registry.DefaultModelRegistry.fireAction(DefaultModelRegistry.java:480)
        ... 67 more

and looked in the source.
Apparently the DSL has changed and we should use "useProguard" instead of "minifyEnabled".

**Comment by zlwu** [comment]
Great, thanks. Not sure if the functionality is the same as well.

**Comment by proppy** [comment]
Let's keep this open to update the sample to the correct syntax.

**Comment by ph0b** [comment]
*useProguard* and *minifyEnabled* are indeed actually different options now. *minifyEnabled* seems to now refer to a new shrinker that is different than proguard (cf. http://tools.android.com/tech-docs/new-build-system)

Back to the original issue, I could set minifyEnabled to true as long as *useProguard* was defined (to false or true - it doesn't matter):

    android.buildTypes {
        debug {
            minifyEnabled = true
            useProguard = false
        }
        release {
            minifyEnabled = true
            useProguard = true
            proguardFiles.add(file('proguard-rules.txt'))
        }
    }

@zlwu can you file a bug on the [tools bug tracker](https://code.google.com/p/android/issues/entry?template=Tools bug report), to report the Null-pointer exception happening because *useProguard* isn't instantiated when not configured?

**Comment by zlwu** [comment]
Filed a new bug here https://code.google.com/p/android/issues/detail?id=199876
Thanks for your help.

**Comment by proppy** [comment]
Fixed with `gradle-experimental:0.6.0-alpha9`, we should make sure #144 update the sample with the right syntax.

**Comment by ggfan** [comment]
this one seems to be addressed, close it for now; re-open it otherwise please.
